pkg/controller/builder.go

```diff
@@ -250,3 +253,38 @@ func UpdateDeploymentProxyHook(pc clients.ProxyClientInterface) deploymentcontro
 		return nil
 	}
 }
+
+func UpdateDeploymentFeatureGatesHook(fa featuregates.FeatureGateAccess) deploymentcontroller.DeploymentHookFunc {
+	return func(_ *operatorv1.OperatorSpec, deployment *appsv1.Deployment) error {
+		klog.FromContext(context.Background()).WithName("builder").V(0).Info("FeatureGatesHook updating environment", "deployment", deployment.Name)
+
+		featuregates, err := fa.CurrentFeatureGates()
+		if err != nil {
+			return fmt.Errorf("error getting featuregates.config.openshift.io/cluster: %w", err)
+		}
+
+		vars := []corev1.EnvVar{{Name: "FEATURE_GATES"}}
+		if featuregates.Enabled(features.FeatureGateNewOLM) {
+			vars[0].Value = string(features.FeatureGateNewOLM)
+		}
+
+		var errs []error
+		for i := range deployment.Spec.Template.Spec.InitContainers {
+			err = setContainerEnv(&deployment.Spec.Template.Spec.InitContainers[i], vars)
```

Comment:
As mentioned in the description, this logic is currently using the same approach as with proxy config - set appropriate env vars in containers. If we wanted to move feature gates to args, I think we'd have to add handling to upstream first.